src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,61 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = createArray(currentSize);
 
+    private T[] createArray(int size) {
+        return (T[]) new Object[size];
+    }
+
+    public void push(T value) {
+        table = grow();
+        table[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return table[size - 1];
     }
 
     public T pop() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        T value = peek();
+        table[size - 1] = null;
+        size--;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private T[] grow() {
```

Comment:
This name does not describe what this method do. In addition, there is no need to return the table from this method, you can make the necessary changes inside it, and then simply refer to this table in another method.

Reply:
ok, done